core/src/test/java/com/linecorp/armeria/client/logging/ContentPreviewingClientTest.java

```diff
@@ -0,0 +1,139 @@
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.logging.ContentPreviewer;
+import com.linecorp.armeria.common.logging.ContentPreviewerFactory;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.encoding.EncodingService;
+import com.linecorp.armeria.testing.junit.server.ServerExtension;
+
+class ContentPreviewingClientTest {
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.service("/", (ctx, req) -> HttpResponse.from(
+                    req.aggregate()
+                       .thenApply(aggregated -> {
+                           final ResponseHeaders responseHeaders =
+                                   ResponseHeaders.of(HttpStatus.OK,
+                                                      HttpHeaderNames.CONTENT_TYPE, MediaType.PLAIN_TEXT_UTF_8);
+                           return HttpResponse.of(responseHeaders,
+                                                  HttpData.ofUtf8("Hello " + aggregated.contentUtf8() + '!'));
+                       })));
+            sb.decorator(delegate -> new EncodingService(delegate, unused -> true, 1));
+        }
+    };
+
+    @RepeatedTest(10)
+    void decodedContentPreview() {
+        final WebClient client = WebClient.builder(server.uri("/"))
+                                          .decorator(DecodingClient.newDecorator())
+                                          .decorator(ContentPreviewingClient.builder()
+                                                                            .contentPreview(100)
+                                                                            .newDecorator())
+                                          .build();
+        final RequestHeaders headers = RequestHeaders.of(HttpMethod.POST, "/",
+                                                         HttpHeaderNames.CONTENT_TYPE, "text/plain");
+
+        final ClientRequestContext context;
+        try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
+            final AggregatedHttpResponse res = client.execute(headers, "Armeria").aggregate().join();
+            assertThat(res.contentUtf8()).isEqualTo("Hello Armeria!");
+            assertThat(res.headers().get(HttpHeaderNames.CONTENT_ENCODING)).isEqualTo("gzip");
+            context = captor.get();
+        }
+
+        final RequestLog requestLog = context.log().whenComplete().join();
+        assertThat(requestLog.requestContentPreview()).isEqualTo("Armeria");
+        assertThat(requestLog.responseContentPreview()).isEqualTo("Hello Armeria!");
+    }
+
+    @Test
+    void contentPreviewIsDecodedInPreviewer() {
```

Comment:
What is this testing? Or mostly for documentation of ordering semantics?

Reply:
Just wanted to show that the preview can be different depends on the orders it attached.

Reply:
Probably good to add a comment about that

Reply:
Added. 😉